src/server.js

```diff
@@ -688,9 +687,7 @@ function onCompletion(params) {
 
 		const start = diagnostic.range.start;
 
-		const rule = String(
-			(DiagnosticCode.is(diagnostic.code) ? diagnostic.code.value : diagnostic.code) || '',
```

Comment:
Check is no longer necessary as per https://github.com/microsoft/vscode-languageserver-node/issues/601

Reply:
I don't understand that code but it looks like there might potentially be a behavior change here. Previously it would fallback to empty string and then the code later would remove empty strings from the Set (`thisLineRules.delete('');`). Now, if we add undefined value here, it won't be removed.

I haven't actually checked if that can have some impact on the behavior later on.

Reply:
Good eye, I simply missed the `|| ''` in the original line. I've pushed up a fix that restores the empty string fallback case.